Add a documentation for pdk providers (#204)
common-fate/docs: docs/common-fate/deploying-common-fate/v2/getting-started.md

> import Tabs from "@theme/Tabs";

Reviewer: Please add a section here to walk through a schema for a provider, and show users how to query for a Provider's schema.

This is an example command which I personally use to inspect the schema:

```
curl https://api.registry.commonfate.io/v1alpha1/providers/common-fate/aws/v0.2.0
```

We should include this command in the docs and show an example output:
```json
{"cfnTemplateS3Arn":"registryassetsstack-assetspackagedproviderassetsb-1m7ey6317uxfh/common-fate/aws/v0.2.0/cloudformation.json","lambdaAssetS3Arn":"registryassetsstack-assetspackagedproviderassetsb-1m7ey6317uxfh/common-fate/aws/v0.2.0/handler.zip","name":"aws","publisher":"common-fate","schema":{"$id":"https://registry.commonfate.io/schema/common-fate/aws/v1","$schema":"https://schema.commonfate.io/provider/v1alpha1","config":{"identity_store_id":{"description":"the AWS SSO identity store ID","secret":false,"type":"string"},"instance_arn":{"description":"the AWS SSO instance ARN","secret":false,"type":"string"},"region":{"description":"the AWS SSO instance region","secret":false,"type":"string"},"sso_role_arn":{"description":"The ARN of the AWS IAM Role with permission to administer SSO","secret":false,"type":"string"}},"meta":{"framework":"0.4.0"},"resources":{"loaders":{"fetch_groups":{"title":"fetch_groups"},"fetch_org_structure":{"title":"fetch_org_structure"},"fetch_permission_sets":{"title":"fetch_permission_sets"},"fetch_users":{"title":"fetch_users"}},"types":{"Account":{"data":{"id":""},"type":"object"},"AccountAssignment":{"data":{"id":""},"type":"object"},"Group":{"data":{"id":""},"type":"object"},"GroupMembership":{"data":{"id":""},"type":"object"},"ManagedPolicy":{"data":{"id":""},"type":"object"},"ManagedPolicyAttachment":{"data":{"id":""},"type":"object"},"OrgUnit":{"data":{"id":""},"type":"object"},"PermissionSet":{"data":{"id":""},"type":"object"},"User":{"data":{"id":""},"type":"object"}}},"targets":{"Account":{"properties":{"account":{"description":"the AWS account to grant access to","id":"","resource":"Account","title":"Account"},"permission_set":{"description":"the AWS permission set to grant access to","id":"","resource":"PermissionSet","title":"Permission Set"}},"type":"object"}}},"version":"v0.2.0"}
```

The schema section should go through each section in the schema and explain what it is used for.

> import Tabs from "@theme/Tabs";
> import TabItem from "@theme/TabItem";

Reviewer: Remove the MDX from here - these will show up in the website text

```suggestion
```

> # Provider 

Reviewer: ```suggestion
# Provider Development Kit (PDK)
```

> Provider is a Python-based service that enables you to create just-in-time access requests for managing permissions across a range of cloud providers, SaaS applications, and CI/CD platforms. 

Reviewer: some grammar/wording fixes

```suggestion
The Provider Development Kit (PDK) is a Python-based framework to build Providers for Common Fate. Providers connect Common Fate with any cloud provider or SaaS application with an API for managing permissions.
```

With the Provider framework, you have access to a standardized API for granting or revoking access. This allows you to build your own provider that is tailored to meet the unique requirements of your company. 

> 

Reviewer: Remove or reword this, remember that the audience for this is "Common Fate administrators looking to deploy a Provider", not "Common Fate developers who want to build their own Provider"

There are also Common Fate built Providers such as AWS SSO, Okta, Google Groups which you can instantly setup in your environment and create workflows with fine-grained permissions.

![diagram of Provider framework](../../../../static/img/pdk/pdk-overview.png)

# Provider Registry 

> A Provider registry is a centralized location where Access Providers are stored, managed, and distributed. It's a server-side application that stores and serves information about Providers. When a developer creates a new Provider, they can push it to the registry, making it available for other users to pull and use in their own environment.

Reviewer: Reword this - we only have one provider registry in production. Also include the URL for the provider registry: `https://api.registry.commonfate.io`.

e.g.

"Providers are published to the Provider Registry. The Provider Registry is a centralised service which allows Common Fate users to discover and share Providers with other users. The Provider Registry is hosted by Common Fate and is available at `https://api.registry.commonfate.io`."

> Provider registries can be either public or private. A public registry, such as the AWS SSO, allows anyone to use this Access Provider in their own environment. In contrast, a private registry is only accessible to authorized users, typically within a specific organization.

Reviewer: We don't have private registries or providers yet, so this won't make sense. We can remove it for now.

Note that when we do, it'll still be a public registry, but with private providers. Similar to how NPM works.

Organizations can also host their own Access Providers securely using Provider Registry.  

Provider Registry is hosted at https://api.registry.commonfate.io

:::note

> Provider Registry is still in alpha version and will continue to improve.

Reviewer: wording/grammar fix

```suggestion
The Provider Registry is currently in alpha. Registry APIs are subject to change.
```

:::

> # Provider Development Kit (PDK)
> 
> You can get started to creating your own Access Provider using our [Python Provider Development Kit](https://github.com/common-fate/commonfate-provider-core). 

Reviewer: remove this for now, the PDK will have it's own getting started guide. Users won't be able to follow that link anyway, as the README on that repo is not friendly for getting started

```suggestion
```
